lib/Controller/ReportController.php

```diff
@@ -70,4 +72,29 @@ public function __construct(
 	public function getReport() {
 		return new ReportResponse('', '', $this->reportDataCollector->getReportJson());
 	}
+
+	/**
+	 * API-Handler which is used to get web config-report from cli. Auth-Token
+	 * is generated by the cli-app for onetime use.
+	 *
+	 * @NoCSRFRequired
+	 * @NoAdminRequired
+	 * @PublicPage
+	 *
+	 * @return ReportResponse|DataResponse with the report
+	 */
+	public function fromCli() {
+		// token generated by cli app
+		$actualToken = $this->config->getAppValue('configreport', 'token');
+		$passedToken = $this->request->getParam('token');
+
+		if (($actualToken !== $passedToken) || (empty($passedToken) || empty($actualToken))) {
+			return new DataResponse([], Http::STATUS_FORBIDDEN, []);
+		}
+
+		$resp = new ReportResponse('', '', $this->reportDataCollector->getReportJson());
+		$this->config->deleteAppValue('configreport', 'token');
```

Comment:
error handling here. response forbidden if fails to delete app value ?

Reply:
if the deletion can fail, we probably should include some timing data in the token to ensure the token is valid only for a few seconds.

Reply:
very good point, probably separate table so this can be flexibly extended. some other apps like wopi etc have similar handling

Reply:
and cronjob to clean failed tokens… but ye rarely apps do that and TTL should be enough.

Reply:
I'd keep the info within the token itself, no need to add more complexity. Taking into account that knowing when the token will expire is irrelevant (I think).
```
$token = [
  't' => $randombytes,
  'e' => time() + 5,
];
$finalToken = base64_encode(json_encode($token));
```
That should be good enough for a token with expiration. We just need to check that `$finalToken` is the same as the one stored in the DB and that the expiration time is valid. Even if someone tampers the token, we still have the original, so changing the expiration date won't validate.

Note that the token will be overwritten the next run of the command, so there is no real need for a cron job

Reply:
good idea to keep it simple